src/csharp/Grpc.Core/Internal/BatchContextSafeHandle.cs

```diff
@@ -106,6 +113,25 @@ public byte[] GetReceivedMessage()
             return data;
         }
 
+        public bool GetReceivedMessageNextSlicePeek(out Slice slice)
+        {
```

Comment:
nit: any reason not to put the body of this function directly into `TryGetNextSlice` ?